demo/src/app/pages/demo/components/custom-column.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, ViewChild, ElementRef } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { Subscription } from 'rxjs/Subscription';
+import 'rxjs/add/observable/combineLatest';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/startWith';
+
+import { Cell, DefaultCellType } from '../../../../../../src/ng2-smart-table.module';
+
+@Component({
+  template: `
+    From: <input [ngClass]="inputClass"
+            #firstInput
+            class="form-control short-input"
+            [value]="firstVal"
+            [name]="cell.getId()"
+            [disabled]="!cell.isEditable()"
+            [placeholder]="cell.getTitle()"
+            (click)="onClick.emit($event)"
+            (keydown.enter)="onEdited.emit($event)"
+            (keydown.esc)="onStopEditing.emit()">
+    To: <input [ngClass]="inputClass"
+            #secondInput
+            class="form-control short-input"
+            [value]="secondVal"
+            [name]="cell.getId()"
+            [disabled]="!cell.isEditable()"
+            [placeholder]="cell.getTitle()"
+            (click)="onClick.emit($event)"
+            (keydown.enter)="onEdited.emit($event)"
+            (keydown.esc)="onStopEditing.emit()">
+    `,
+    styles: [`
+      .short-input{
+        width: 30%;
+      }
+    `]
+})
+export class CustomColumnComponent extends DefaultCellType{
```

Comment:
not exactly, we should have the following structure
DefaultCellType (renamed in future to DefaultEditor) - genenaral type, has general type logic implementation
Editor - interface which contains required methods to be declared by each editor
CustomEditor, CompleterEditor, SelectEditor - all our editors which will (or will not) extend from DefaultEditor and must implement Editor interface.

In this case we can ensure that even custom editor has required to any editor methods and we don't get unexpected errors.